lib/src/presentation/view/home/widgets/home_banner.dart

```diff
@@ -7,19 +7,33 @@ import 'package:coofix/src/presentation/view/home/widgets/custom_marquee.dart';
 import 'package:flutter/material.dart';
 import 'package:flutter_bloc/flutter_bloc.dart';
 
-class HomeBanner extends StatelessWidget {
-  const HomeBanner({super.key,});
+class HomeBanner extends StatefulWidget {
+  const HomeBanner({
+    super.key,
+  });
+
+  @override
+  State<HomeBanner> createState() => _HomeBannerState();
+}
+
+class _HomeBannerState extends State<HomeBanner> {
+  @override
+    void didChangeDependencies() {
+    context.read<BannerBloc>().add(const BannerEvent.getBanner());
```

Comment:
Why bloc event called inside didChangeDependencies method. 
Please make sure the proper flow of event calling.